app/concepts/coupon/policy/apply.rb

```diff
@@ -0,0 +1,7 @@
+class Coupon::Policy::ApplyGuard
+  include Uber::Callable
+
+  def call(_ctx, params:, **)
+    params[:order].present? && params[:order].is_a?(Order)
```

Comment:
do we need to check order class?
I think you should check if the order belongs to the current_user

Reply:
User assigns to order only when checkout process started.
Coupon works just like on `https://www.udemy.com/`. It can be applied to anonymous order

Reply:
ok